starter/src/main/java/com/vivy/springdoc/OpenAPIAutoConfiguration.java

```diff
@@ -32,4 +50,66 @@ public OpenAPI bearerOpenAPI() {
                 ))
                 .addSecurityItem(new SecurityRequirement().addList("bearer-jwt", List.of("read", "write")));
     }
+
+    @Bean
+    @ConditionalOnBean(OpenAPI.class)
+    public ApplicationRunner applicationRunner(Environment environment) {
+
+        String APPLICATION_PROPERTIES_PREFIX = "openapi-boot-starter.swagger-ui.api-docs.";
+        DefaultUriBuilderFactory uriFactory = new DefaultUriBuilderFactory();
+
+        return new ApplicationRunner() {
+            @Override
+            public void run(ApplicationArguments args) throws Exception {
+                int apiDocPort = environment.getProperty("local.server.port", Integer.class, -1);
+                String openApiPath = getEnvVariable("openApiEndpoint.path", String.class, "/v3/api-docs");
+                UriBuilder uriBuilder = uriFactory.builder().scheme("http").host("localhost").path(openApiPath).port(apiDocPort);
```

Comment:
you don't really need to build an url this way - it's too verbosive - just create builder from entire final url and change only the port

Reply:
hm im not sure if its possible to create a builder from the entire url. You can perhaps create a URI object ```new URI(url)``` but then you will not be able to manipulate the path, thats why i have the builder as you can see i use this object later on to retrive the url with out the path inorder to replace it. So in short i use the builder since i can extract two urls from it, one url with port and endpoint "http://localhost:38383/v3/api-doc" and the second one "http://localhost" inorder to replace the base url in the document. This way i dont create two times similar url.